httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/InternalAbstractHttpAsyncClient.java

```diff
@@ -219,6 +220,7 @@ protected <T> Future<T> doExecute(
                 setupContext(clientContext);
 
                 final HttpHost resolvedTarget = target != null ? target : RoutingSupport.determineHost(request);
+                Args.notNull(resolvedTarget, "Target host for the request");
```

Comment:
@arturobernalg This is not right. The target may be null if the request URI cannot be parsed. It should be up to the specific protocol handler to decide if such a request can be executed. H2 will likely reject it but HTTP/1.1 may accept it